src/crypto/crypto.js

```diff
@@ -152,83 +151,141 @@ export default {
+   * @param {Object}                 publicParams (ECC only) public params, needed to format some private params
+   * @returns { read: Number, params: Object } number of read bytes plus the key parameters referenced by name
+   */
+  parsePrivateKeyParams: function(algo, bytes, publicParams) {
+    let read = 0;
+    switch (algo) {
+      //   Algorithm-Specific Fields for RSA secret keys:
+      //       - multiprecision integer (MPI) of RSA secret exponent d.
+      //       - MPI of RSA secret prime value p.
+      //       - MPI of RSA secret prime value q (p < q).
+      //       - MPI of u, the multiplicative inverse of p, mod q.
+      //   Algorithm-Specific Fields for Elgamal and DSA secret keys:
+      //        - MPI of Elgamal secret exponent x.
+      case enums.publicKey.rsaEncrypt:
+      case enums.publicKey.rsaEncryptSign:
+      case enums.publicKey.rsaSign:
+      case enums.publicKey.dsa:
+      case enums.publicKey.elgamal: {
+        const params = {};
+        const paramNames = this.getPrivateKeyParamNames(algo);
+        paramNames.forEach(name => {
+          const param = util.readMPI(bytes.subarray(read)); read += param.length + 2;
+          params[name] = param;
+        });
+        return { read: read, params };
+      }
+      //   Algorithm-Specific Fields for ECDSA, EDDSA or ECDH secret keys:
+      //       - MPI of an integer representing the secret key.
+      case enums.publicKey.ecdsa:
+      case enums.publicKey.ecdh: {
+        const curve = new Curve(publicParams.oid);
+        let d = util.readMPI(bytes.subarray(read)); read += d.length + 2;
+        d = util.padToLength(d, curve.payloadSize);
+        return { read, params: { d } };
+      }
+      case enums.publicKey.eddsa: {
+        let seed = util.readMPI(bytes.subarray(read)); read += seed.length + 2;
+        seed = util.padToLength(seed, 32);
+        return { read, params: { seed } };
+      }
       default:
         throw new Error('Invalid public key encryption algorithm.');
     }
   },
 
+  /**
+   * Convert private params to MPI and serializes them in the proper order
+   * @param {module:enums.publicKey}  algo The public key algorithm
+   * @param {Object}                  params The private key parameters indexed by name
+   * @returns {Uint8Array}  The array containing the MPIs
+   */
+  serializePrivateKeyParams: function(algo, params) {
+    const paramNames = this.getPrivateKeyParamNames(algo);
+    return serializeKeyParams(params, paramNames);
+  },
+
+  /**
+   * Convert public params to MPI and serializes them in the proper order
+   * @param {module:enums.publicKey}  algo The public key algorithm
+   * @param {Object}                  params The private key parameters indexed by name
+   * @returns {Uint8Array}  The array containing the MPIs
+   */
+  serializePublicKeyParams: function(algo, params) {
+    const paramNames = this.getPublicKeyParamNames(algo);
+    return serializeKeyParams(params, paramNames);
```

Comment:
Rather than abstracting this away, I would make an explicit switch statement here, with something like:
```js
switch (algo) {
  case rsa:
    const { n, e } = params;
    return util.concatUint8Array([n, e].map(util.uint8ArrayToMpi));
  // ...
}
```
I know it's longer, but then you can get rid of `getPublicKeyParamNames`, and it's more flexible for symmetric keys, for example (they don't have to be MPIs).
If we want to be a bit adventurous, you could also do:
```js
switch (algo) {
  case rsa:
  case dsa:
  case elgamal:
    return util.concatUint8Array(Object.values(params).map(util.uint8ArrayToMpi));
  // ...
}
```
Since Object order is nowadays defined in the ECMAscript spec. Even in old browsers, this should be perfectly safe if we have no numeric keys etc.